library/lvm_gensym

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python
+
+import subprocess
+import StringIO 
+import json
+from ansible.module_utils.basic import *
+
+def get_os():
+	cat_cmd = subprocess.check_output('cat /etc/os-release', shell = True)
+	buf =  StringIO.StringIO(cat_cmd)
+	
+	for line in buf:
+		if not line.find('ID='):
+			name = line[3:]
+			
+	name = name.replace('\n', '').replace('"', '')
+	return name 
+
+def get_host():
+	host_buf = subprocess.check_output('hostname', shell = True)
+	buf = host_buf.replace('\n', '')
+	buf = buf.replace('.', '_')
+	return buf 
+
+def test_default_name(default_name):
+	vgs_cmd = subprocess.check_output("sudo vgs | awk '{print $1}'", shell = True)
+	buf = vgs_cmd.split('\n')
+	buf = filter(None, buf)
+
+	for line in buf[1:]:
+		if not line.lower().find(default_name.lower()):
+			return False
+	return True 
+ 
+def generate_name(test_str, counter):
+	if (counter == 1):
+		return test_str + '_' + str(counter)
+
+	else:
+		return test_str[:-2] + '_' + str(counter)
+
+def test_invalid_name(test_str, counter):
+	test_str = generate_name(test_str, counter)
+
+	if (test_default_name(test_str)):
+		return test_str 
+	else:
+		counter += 1 
+		return test_invalid_name(test_str, counter)
+
+def main():
+	module = AnsibleModule(argument_spec={})
+	os_name = get_os()
+	host_name = get_host()
+	default_name = os_name + '-' + host_name
+	unique_flag = test_default_name(default_name)
+	counter = 1
+
+	result = {}
+
+	if (unique_flag):
+		result['name'] = default_name
+		module.exit_json(**result)
+	else:
```

Comment:
You should probably replace this with a loop of the form:

```python
counter = 0
while unique_flag is False:
    counter += 1
    default_name = ...
    unique_flag = ...
```

Then it will be able to handle several vgs already existing using similar gensyms.